webclient/projekt/views.py

```diff
@@ -969,18 +972,24 @@ def archivovat(request, ident_cely):
             status=403,
         )
     if request.method == "POST":
-        projekt.create_transaction(request.user, PROJEKT_USPESNE_ARCHIVOVAN)
-        projekt.set_archivovany(request.user)
-        projekt.close_active_transaction_when_finished = True
-        projekt.save()
-        for item in projekt.casti_dokumentu.all():
-            item: DokumentCast
-            if item.dokument.doi and item.dokument.stav == D_STAV_ARCHIVOVANY:
-                item.dokument.doi_update()
-        for item in projekt.samostatne_nalezy.all():
-            item: SamostatnyNalez
-            if item.igsn and item.stav == SN_ARCHIVOVANY:
-                item.igsn_update()
+        fedora_transaction = projekt.create_transaction(request.user, PROJEKT_USPESNE_ARCHIVOVAN)
+        try:
+            projekt.set_archivovany(request.user)
+            for item in projekt.casti_dokumentu.all():
+                item: DokumentCast
+                if item.dokument.doi and item.dokument.stav == D_STAV_ARCHIVOVANY:
+                    item.dokument.doi_update()
+            for item in projekt.samostatne_nalezy.all():
+                item: SamostatnyNalez
+                if item.igsn and item.stav == SN_ARCHIVOVANY:
+                    item.igsn_update()
+            projekt.close_active_transaction_when_finished = True
+            projekt.save()
```

Comment:
Nelze přesunout na konec kvůli správné aktualizaci metadat...

Reply:
Přesunuto.

Reply:
Proč?

Reply:
Stejný důvod jako jinde - pokud se projet nejdříve neuloží, nebudou se myslím správně vyhodnocovat podmínky při serializaci DOI/IGSN. Jestli si dobře pamatuju testování při implementaci, tak právě kvůli tomu se to muselo posouvat na správná místa.